build.rb

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/ruby 
+#
+#  -couch <type> couchdb1.6, couchdb2.0, cloudantSAAS, cloudantlocal - default is couchdb1.6
+
+#  -plaform <platfrom> java | android default is Java
+#  -D* gets passed into build.
+#  
+#
+
+params = { :d_options => Array.new }
+arg_is_value = false 
+prev_arg = nil
+
+
+ARGV.each do |arg|
+
+	puts arg
+
+ if arg.start_with?("-D")
+    params[:d_options].push(arg)
+ 	next
+ end
+
+ #prcoess arguments into a hash
+ unless arg_is_value 
+ 	params[arg[1,arg.length] ] = nil
+ 	$prev_arg = arg[1,arg.length] 
+ 	arg_is_value = true
+ else
+ 	params[$prev_arg] = arg
+ 	arg_is_value = false 
+ end
+
+end
+
+#apply defaults
+params["platform"] = "java" unless params["platform"] 
+params["couch"] = "couchdb1.6" unless params["couch"]
+
+
+#launch docker
+puts "Starting docker container #{$couch}"
+puts "Commandline: docker run -p 5984:5984 -d --name 'couchdb' #{params["couch"]}"
```

Comment:
If you want to output this string (it might be useful in a verbose/debug mode), then set a variable so you don't have to repeat yourself in the `system` call